src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/PolymorphicTests.CustomTypeHierarchies.cs

```diff
@@ -1762,12 +1880,34 @@ public async Task Peano_Deserialization(int expectedSize, string json)
             await TestMultiContextDeserialization<Peano>(json, expected);
         }
 
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(3)]
+        [InlineData(4)] // initial Write/ReadStack size
+        [InlineData(5)]
+        [InlineData(63)]
+        [InlineData(64)] // initial Write/ReadStack size * 16
+        [InlineData(65)]
+        [InlineData(150)]
```

Comment:
* $150$ might be bringing the serializer close to SO, at least for certain platforms.
* Let's stick with smaller 2^k + 1 values which test resize operations
```suggestion
        [InlineData(5)] // 1st Write/ReadStack resize
        [InlineData(17)] // 2nd Write/ReadStack resize
        [InlineData(33)] // 3rd Write/ReadStack resize
        [InlineData(65)] // 4th Write/ReadStack resize
```